regression-test/suites/backup_restore/test_backup_restore_reserve_dynamic_partition_true.groovy

```diff
@@ -0,0 +1,114 @@
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+suite("test_backup_restore_dynamic_partition_reserve_true", "backup_restore") {
+    String repoName = "test_backup_restore_dynamic_partition_reserve_true_repo"
+    String dbName = "backup_restore_dynamic_partition_reserve_true_db"
+    String tableName = "dynamic_partition_reserve_true_table"
+
+    def syncer = getSyncer()
+    syncer.createS3Repository(repoName)
+
+    sql "CREATE DATABASE IF NOT EXISTS ${dbName}"
+    sql "DROP TABLE IF EXISTS ${dbName}.${tableName}"
+    sql """
+        CREATE TABLE ${dbName}.${tableName}  (
+            `id` LARGEINT NOT NULL,
+            `sdate` DATE
+        )
+        PARTITION BY RANGE(sdate)()
+        DISTRIBUTED BY HASH(`id`) BUCKETS 2
+        PROPERTIES
+        (
+            "dynamic_partition.enable" = "true",
+            "dynamic_partition.time_unit" = "YEAR",
+            "dynamic_partition.start" = "-50",
+            "dynamic_partition.end" = "5",
+            "dynamic_partition.prefix" = "p",
+            "dynamic_partition.create_history_partition"="true",
+            "replication_num" = "1"
+        )
+        """
+
+    sql """
+        INSERT INTO ${dbName}.${tableName} VALUES
+        (1,"2023-11-01"),
+        (2,"2023-11-01"),
+        (3,"2023-11-01"),
+        (4,"2023-11-01"),
+        (5,"2023-11-01"),
+        (1,"2023-11-02"),
+        (2,"2023-11-02"),
+        (3,"2023-11-02"),
+        (4,"2023-11-02"),
+        (5,"2023-11-02"),
+        (1,"2024-11-01"),
+        (2,"2024-11-01"),
+        (3,"2024-11-01"),
+        (4,"2024-11-01"),
+        (5,"2024-11-01"),
+        (1,"2024-11-02"),
+        (2,"2024-11-02"),
+        (3,"2024-11-02"),
+        (4,"2024-11-02"),
+        (5,"2024-11-02")
+        """
+
+    def result = sql "SELECT * FROM ${dbName}.${tableName}"
+    assertEquals(result.size(),20)
+
+    String snapshotName = "test_backup_restore_dynamic_partition_reserve_true_snapshot"
+    sql """
+        BACKUP SNAPSHOT ${dbName}.${snapshotName}
+        TO `${repoName}`
+        ON (${tableName})
+    """    
+
+    while (!syncer.checkSnapshotFinish(dbName)) {
+        Thread.sleep(3000)
+    }
+    def snapshot = syncer.getSnapshotTimestamp(repoName, snapshotName)
+    assertTrue(snapshot != null)
+
+    sql "TRUNCATE TABLE ${dbName}.${tableName}"
+
+    sql """
+        RESTORE SNAPSHOT ${dbName}.${snapshotName}
+        FROM `${repoName}`
+        ON (`${tableName}`)
+        PROPERTIES
+        (
+            "backup_timestamp" = "${snapshot}",
+            "reserve_dynamic_partition_enable" = "true",
+            "replication_num" = "1"
+        )
+    """
+    
+    // def restore_properties = sql "SHOW CREATE TABLE ${dbName}.${tableName}"
+
+    // assertTrue(restore_properties.indexOf("\"dynamic_partition.enable\" = \"true\"") != -1)
```

Comment:
The result of `SHOW CREATE TABLE` contains a row with two columns, like: 

```
| Table                                 | Create Table 
| dynamic_partition_reserve_true_table  | CREATE TABLE XXXXX
```

To access the first row and second column, use `restore_properties[0][1]`